### ISSUE TITLE
VisualScript crashes when moving ui_left from project settings input map to visual editor area

### ISSUE THREAD

**Issue by trupljan** [godotengine/godot#60043]
### Godot version

Godot_v3.4.4-stable_x11.64

### System information

Linux, Ubuntu 20.04 LTS, NVIDIA GTX 1650 Super, GLES3

### Issue description

I have tried VisualScript for first time to check if it is any useful, and I found several ways to crash it during my fooling around. This is one of the bugs I found. When I tried to figure out how to add input node for left key, I got an idea to open input map in project settings and move ui_left from here to the area of editor. It crashed instantly. While unprobable, I expected it might actually work. Anyway, while the dialog looks modal, it did not prevented me from moving the element to backround and it crashed instead.

```
ERROR: FATAL: Index p_index = 0 is out of bounds (((Vector<T> *)(this))->_cowdata.size() = 0).
   at: operator[] (./core/vector.h:49)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v3.4.4.stable.official (419e713a29f20bd3351a54d1e6c4c5af7ef4b253)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x430c0) [0x7fd02a99b0c0] (??:0)
[2] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x2aec4e1] (??:0)
[3] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0xc3bbaf] (??:0)
[4] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0xc13657] (??:0)
[5] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x2b77c39] (??:0)
[6] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x2b6ab94] (??:0)
[7] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1df9614] (??:0)
[8] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1d88d0f] (??:0)
[9] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1dbe10d] (??:0)
[10] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1dc0225] (??:0)
[11] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1dc038b] (??:0)
[12] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0xc62695] (??:0)
[13] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x2b77c39] (??:0)
[14] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x2b6ab94] (??:0)
[15] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1d967ed] (??:0)
[16] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x1da81d9] (??:0)
[17] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x98029b] (??:0)
[18] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x9814a9] (??:0)
[19] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x98a922] (??:0)
[20] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x94719c] (??:0)
[21] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fd02a97c0b3] (??:0)
[22] /home/tester/godot/Godot_v3.4.4-stable_x11.64() [0x95ae1e] (??:0)
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

1. Create new project (OpenGL ES 3.0)
2. Click 3D Scene to create Spatial node
3. Add CSGBox as a child to Spatial node
4. Attach new VisualScript CSGBox.vs to the box
5. Click menu: Project > Project settings... to open Project Settings
6. Click Input Map
7. Click on ui_left and move it to the area of Visual Script editor
8. Crash!

**Comment by Calinou** [comment]
@trupljan Can you try to get a full backtrace by running a [debug build](https://github.com/Calinou/godot-debug-builds) from a terminal? This will tell you where the issue comes from.

**Comment by trupljan** [comment]
There it is:
```
Godot Engine v3.4.4.stable.custom_build.419e713a2 - https://godotengine.org
OpenGL ES 3.0 Renderer: NVIDIA GeForce GTX 1650 SUPER/PCIe/SSE2
OpenGL ES Batching: ON
/lib/x86_64-linux-gnu/libasound.so.2: undefined symbol: snd_dlpath
 
ERROR: FATAL: Index p_index = 0 is out of bounds (((Vector<T> *)(this))->_cowdata.size() = 0).
   at: operator[] (./core/vector.h:49)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v3.4.4.stable.custom_build (419e713a29f20bd3351a54d1e6c4c5af7ef4b253)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x430c0) [0x7f0d0b0570c0] (??:0)
[2] VectorWriteProxy<Variant>::operator[](int) (/opt/godot-3.4.4-stable/./core/vector.h:49 (discriminator 7))
[3] Array::operator[](int) (/opt/godot-3.4.4-stable/core/array.cpp:75)
[4] VisualScriptEditor::drop_data_fw(Vector2 const&, Variant const&, Control*) (/opt/godot-3.4.4-stable/modules/visual_script/visual_script_editor.cpp:2280)
[5] MethodBind3<Vector2 const&, Variant const&, Control*>::call(Object*, Variant const**, int, Variant::CallError&) (/opt/godot-3.4.4-stable/./core/method_bind.gen.inc:2303 (discriminator 26))
[6] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/opt/godot-3.4.4-stable/core/object.cpp:918 (discriminator 1))
[7] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/opt/godot-3.4.4-stable/core/object.cpp:848)
[8] Control::drop_data(Vector2 const&, Variant const&) (/opt/godot-3.4.4-stable/scene/gui/control.cpp:731 (discriminator 6))
[9] Viewport::_gui_drop(Control*, Vector2, bool) (/opt/godot-3.4.4-stable/scene/main/viewport.cpp:1814)
[10] Viewport::_gui_input_event(Ref<InputEvent>) (/opt/godot-3.4.4-stable/scene/main/viewport.cpp:1996)
[11] Viewport::input(Ref<InputEvent> const&) (/opt/godot-3.4.4-stable/scene/main/viewport.cpp:2830 (discriminator 2))
[12] Viewport::_vp_input(Ref<InputEvent> const&) (/opt/godot-3.4.4-stable/scene/main/viewport.cpp:1409)
[13] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (/opt/godot-3.4.4-stable/./core/method_bind.gen.inc:775 (discriminator 12))
[14] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/opt/godot-3.4.4-stable/core/object.cpp:918 (discriminator 1))
[15] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/opt/godot-3.4.4-stable/core/object.cpp:848)
[16] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/opt/godot-3.4.4-stable/scene/main/scene_tree.cpp:280)
[17] SceneTree::input_event(Ref<InputEvent> const&) (/opt/godot-3.4.4-stable/scene/main/scene_tree.cpp:439 (discriminator 6))
[18] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/opt/godot-3.4.4-stable/main/input_default.cpp:462)
[19] InputDefault::flush_buffered_events() (/opt/godot-3.4.4-stable/main/input_default.cpp:711)
[20] OS_X11::process_xevents() (/opt/godot-3.4.4-stable/platform/x11/os_x11.cpp:2356)
[21] OS_X11::run() (/opt/godot-3.4.4-stable/platform/x11/os_x11.cpp:3641)
[22] ../build/godot-3.4.4-stable/bin/godot.x11.tools.64(main+0x125) [0x179e50b] (/opt/godot-3.4.4-stable/platform/x11/godot_x11.cpp:57)
[23] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f0d0b0380b3] (??:0)
[24] ../build/godot-3.4.4-stable/bin/godot.x11.tools.64(_start+0x2e) [0x179e32e] (??:?)
-- END OF BACKTRACE --
================================================================

```

**Comment by akien-mga** [comment]
Fixed by #60048.
